src/basic/uid-range.c

```diff
@@ -214,13 +195,20 @@ int uid_range_load_userns(UidRange **p, size_t *n, const char *path) {
                         if (ferror(f))
                                 return errno_or_else(EIO);
 
-                        return 0;
+                        break;
                 }
                 if (k != 3)
                         return -EBADMSG;
 
-                r = uid_range_add(p, n, uid_base, uid_range);
+                r = uid_range_add(&q, &m, uid_base, uid_range);
```

Comment:
i figure you wanted to change this to uid_range_add_internal() with coalesce=false, no?